lib/Runtime/Library/BoundFunction.cpp

```diff
@@ -136,16 +136,14 @@ namespace Js
         Var newVarInstance = nullptr;
         if (callInfo.Flags & CallFlags_New)
         {
-          if (JavascriptProxy::Is(targetFunction))
-          {
-            JavascriptProxy* proxy = JavascriptProxy::FromVar(targetFunction);
-            Arguments proxyArgs(CallInfo(CallFlags_New, 1), &targetFunction);
-            args.Values[0] = newVarInstance = proxy->ConstructorTrap(proxyArgs, scriptContext, 0);
-          }
-          else
-          {
-            args.Values[0] = newVarInstance = JavascriptOperators::NewScObjectNoCtor(targetFunction, scriptContext);
-          }
+            if (JavascriptProxy::Is(targetFunction))
+            {
+                args.Values[0] = newVarInstance = JavascriptProxy::FromVar(targetFunction);//JavascriptOperators::CreateFromConstructor(RecyclableObject::FromVar(targetFunction), scriptContext);
```

Comment:
This feels wrong... But the proxy FunctionCallTrap method forwards args[0] as the newTarget for the construct trap when called from a boundFunction. Hence need it to be the proxy itself and not "this". A better fix may be to edit FunctionCallTrap.